### PR TITLE
feat : Enhance contest card UI and update related test

### LINKED ISSUE #5998: Contests Enhancements
### **Problem: The Contests Feature Needs a Revamp**

The current Contests feature is simplistic and lacks the engagement needed to be a major success. 

----

**_Current actionable task at hand :_**

1.  **Aesthetics:** Contest cards lack compelling graphics, which makes them less appealing and informative. We need some good graphics for these cards as currently they look like this : 

Image

----

**_Future Work :_** 

1.  **Engagement:** The overall contest experience is very basic, limiting user excitement and participation.

To turn this feature into a success and make contests more engaging for the CircuitVerse community, we need to brainstorm and implement new ideas.

---

### **Requesting Ideas: Making Contests More Engaging**

We're looking for actionable ideas with a clear implementation plan. When you comment with your suggestions, please keep the following constraints in mind:

1.  **No Third-Party Reliance:** Ideas should not depend on external services or APIs.
2.  **Realistic & Maintainable:** Solutions must be realistic and sustainable for an open-source project with a limited budget.
3.  **Stay Lightweight:** We must remember that CircuitVerse's core selling point is its lightweight, easy-to-use simulator. Any additions that would make the platform bloated or difficult to use are not good ideas to consider.

### PR DESCRIPTION
**Pull request by Aakansha-Chavan**
Fixes #5998 

#### Describe the changes you have made in this PR -
Updates the Contest Card UI with a new default image, better styling, and a default description to give it a hackathon type feel. It also updates the system tests to match the new UI strings."

### Screenshots of the UI changes (If any) -
**Before**
User View
user_view_contest

**After**
User View
contest_card_user_view
Admin View
contest_card_admin_view

**Video**
Contest_video_cards

## Code Understanding and AI Usage

**Did you use AI assistance (ChatGPT, Claude, Copilot, etc.) to write any part of this code?**
- [x] No, I wrote all the code myself
- [ ] Yes, I used AI assistance (continue below)

**If you used AI assistance:**
- [ ] I have reviewed every single line of the AI-generated code
- [ ] I can explain the purpose and logic of each function/component I added
- [ ] I have tested edge cases and understand how the code handles them
- [ ] I have modified the AI output to follow this project's coding standards and conventions

**Explain your implementation approach:**
Here, Contest Card had a basic design which was not appealing & attractive , for users to participate in a hackathon so I implemented a (SCSS) design which goes well with the CircuitVerse UI/UX and this will make the Contest feature more engaging and can help the platform to turn new users into potential users.

## Checklist before requesting a review
- [x] I have added proper PR title and linked to the issue
- [x] I have performed a self-review of my code
- [x] **I can explain the purpose of every function, class, and logic block I added**
- [x] I understand why my changes work and have tested them thoroughly
- [x] I have considered potential edge cases and how my code handles them
- [x] If it is a core feature, I have added thorough tests
- [x] My code follows the project's style guidelines and conventions

---

Note: Please check **Allow edits from maintainers** if you would like us to assist in the PR.

## Summary by CodeRabbit

* **New Features**
  * Redesigned contest cards with image region, description, responsive grid, and per-contest actions/modals
  * Persistent submission counters added so entry totals display reliably

* **Improvements**
  * Enhanced hover animations, shadows, spacing, typography, badges, buttons, and layout
  * Updated entry label wording, link behavior, and new translation keys for card content

* **Tests**
  * Updated UI specs to match revised text and per-contest element IDs

### CODE REVIEW

**Comment by Aakansha-Chavan** [comment]
@Asrani-Aman @tachyons  the coderabbit suggestion to move to data-contest-id broke the withdrawal flow in our tests (expected to find text "Withdraw Submission") and this particular change is not directly related to Contest Card UI enhancement , so would you revert (skip) this changes while reviewing this PR ? CI is green

File: db/migrate/20260227150250_add_submissions_count_to_contests.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class AddSubmissionsCountToContests < ActiveRecord::Migration[8.0]
```
Comment (Asrani-Aman) [inline db/migrate/20260227150250_add_submissions_count_to_contests.rb:1]: i guess we don't need this migrations as we are calculating the 
entries contest.submission.count

no db changes required in this PR

**Comment by Aakansha-Chavan** [comment]
@Asrani-Aman the PR is ready for review.

**Comment by tachyons** [comment]
Could you move contest admin features to Avo ?

**Comment by Aakansha-Chavan** [comment]
> Could you move contest admin features to Avo ?

I understand that contest admin management is being handled through Avo now, so I’ll remove the changes related to the old /admins/contests interface and keep this PR focused only on the public/frontend contest UI improvements, like this should be my approach, right @tachyons ?

**Comment by tachyons** [comment]
Yes

**Comment by Aakansha-Chavan** [comment]
@tachyons @Asrani-Aman the PR is ready for review.

**Review by Asrani-Aman — APPROVED** [review]

**Comment by Asrani-Aman** [comment]
@Aakansha-Chavan 
Can you please check 

why this security check stage failed 


```
Run bundle exec bundler-audit --update
Download ruby-advisory-db ...
Cloning into '/home/runner/.local/share/ruby-advisory-db'...
ruby-advisory-db:
  advisories:	1134 advisories
  last updated:	2026-05-23 16:40:42 -0400
  commit:	4e07380b88e53592d0000f8e820fd50c6e3e8437
Name: avo
Version: 3.31.0
CVE: CVE-2026-42205
GHSA: GHSA-qc5p-3mg5-9fh8
Criticality: High
URL: https://github.com/avo-hq/avo/security/advisories/GHSA-qc5p-3mg5-9fh8
Title: Broken Access Control Through Unauthorized Execution of Arbitrary Action Classes Across Resources
Solution: update to '>= 3.31.1'

Name: css_parser
Version: 1.21.1
CVE: CVE-2026-44312
GHSA: GHSA-ff6c-w6qf-7xqc
Criticality: Medium
URL: https://github.com/premailer/css_parser/security/advisories/GHSA-ff6c-w6qf-7xqc
Title: Improper Certificate Validation allows MITM injection of remote CSS content
Solution: update to '~> 1.22.0', '>= 2.1.0'

Vulnerabilities found!
Error: Process completed with exit code 1.

```

**Comment by Aakansha-Chavan** [comment]
@Asrani-Aman yes I will do and inform you.
Thank you 😊

**Comment by Aakansha-Chavan** [comment]
@Asrani-Aman I researched and got to know the security check failed because of 2 vulnerable gems in the project dependencies avo 3.31.0, css_parser 1.21.1 which needs to be updated. Also verified the contest-related specs locally and they are passing successfully. So should I open a separate PR to update the vulnerable gem versions and make the security check pass? or do it here only?
